integr8ly/integreatly-operator
# MGDAPI-5809 Remove legacy monitoring code

# Issue link
JIRA: [MGDAPI-5809](https://issues.redhat.com/browse/MGDAPI-5809)

# What
This PR refactors and removes legacy code from RHOAM's previous monitoring solutions. It still leaves some OO code that is needed to uninstall OO in upgrade scenarios. That code will be removed as part of [MGDAPI-5833](https://issues.redhat.com/browse/MGDAPI-5833).

# Verification steps
# Verification steps
1. Provision a cluster
2. Checkout this PR
3. When the cluster is ready prepare for an OLM install: 
    ```bash
    make cluster/prepare/local
    ```
4. Create a CatalogSource using this [index image](https://quay.io/repository/ckyrillo/managed-api-service-index?tab=tags&tag=1.38.0) (you can also build your own from this PR if you'd prefer):
    ```bash
    cat << EOF | oc create -f -
    apiVersion: operators.coreos.com/v1alpha1
    kind: CatalogSource
    metadata:
      name: rhoam-operators
      namespace: openshift-marketplace
    spec:
      sourceType: grpc
      image: quay.io/ckyrillo/managed-api-service-index:1.38.0
    EOF
    ```
5. Install the RHOAM operator through the OperatorHub
6. Create an RHMI CR:
    ```bash
    IN_PROW=false USE_CLUSTER_STORAGE=true make deploy/integreatly-rhmi-cr.yml
    ```
7. Confirm that the installation finishes without issue
8. Delete the RHMI CR and verify that the uninstallation finishes without issue:
    ```bash
    oc delete rhmi rhoam
    ```
